src/Pate/Verification.hs

```diff
@@ -178,18 +189,23 @@ verifyPairs logAction elf elf' blockMap dcfg pPairs = do
       liftIO . putStr $ ppEquivalenceStatistics stats
       return $ equivSuccess stats
 
+---------------------------------------------
+-- Top-level loop
+
 -- | Verify equivalence of the given pairs, as well as any
 -- resulting pairs that emerge
 runVerificationLoop ::
```

Comment:
Can this get a high-level description of the approach it is taking? It looks like some kind of worklist - is there some required structure to the order in which goals are discharged?